src/main/webapp/components/NoteGridCollection.vue

```diff
@@ -50,6 +50,7 @@ module.exports = {
       this.nextTick(() => this.selected = null)
       // Filter by date, matching terms, and sort with compareFunc
       return this.noteData
+        .map(this.parseData)
```

Comment:
Since every set of `noteData` passed to the `NoteGridCollection` isn't a nested array of notes/numFavoritesInTimespan (like on the profile page where we want to just pass a 1D array of notes to the `NoteGridCollection`), maybe we can introduce a prop to `NoteGridCollection`, something like "trendingData" or something which is a boolean that indicates whether or not this `noteData` is a nested array or 1D array. Then only if `trendingData` is true, we'll map the data using the `parseData` function.

Reply:
Okay, just pushed a commit that seems to implement what you had an mind and still works.

Reply:
Thanks for making the change, it looks great and I think what you've done is better than the suggestion I made to add a prop because this way the user of the component doesn't need to think about what kind of data they're passing, they can just pass one of the expected formats and it just works!